text/0032-react-primary-framework.md

```diff
@@ -0,0 +1,105 @@
+
+Each registered component now has to be React component. In case of other frameworks that is still possible via a converter. In the spirit of the `single-spa` converters we can essentially use the same thing. All that's done is to return a React component much like the following code:
+
+```jsx
+function makeConverter({ onMount, onUpdate, onUnmount }) {
+  const Converter = (props) => {
+    const ref = React.useRef();
+    const lifecycle = React.useRef('bootstrap');
+    const deps = getDependencies(props);
+
+    React.useEffect(() => {
+      const node = ref.current;
+
+      if (node && lifecycle.current === 'mounted') {
+        onUpdate(node, props);
+      }
+    }, deps);
+
+    React.useEffect(() => {
+      const node = ref.current;
+
+      if (node) {
+        onMount(node, props);
+        lifecycle.current = 'mounted';
+
+        return () => {
+          lifecycle.current = 'unmounting';
+          onUnmount(node, props);
+        }
+      }
+    }, []);
+
+    return <div ref={ref} />;
+  };
+}
+```
+
+Such a generic higher-order component can be used to create converters, e.g., to bring Angular components to that world (where React is the primary framework).
+
+Another thing to consider is the use of `single-spa` converters such as `single-spa-react` or `single-spa-angular`. While the former is actually already hidden by default the latter is used, for instance, in the forms frontend module.
+
+It is possible to actually include a shim for this in the tooling. In Webpack it could look like this:
+
+```js
+module.exports = {
+  // webpack config
+  resolve: {
+    alias: {
+      'single-spa-angular': path.resolve(__dirname, 'shims', 'angular'),
+    },
+  },
+};
+```
+
+where the locally placed file would export the same API as `single-spa-angular`, except that it would not export `SingleSpa.LifeCycles`, but rather `React.Component`:
+
+```ts
+export function singleSpaAngular(options) {
+  const lc = createLifecycle(options);
+  return makeConverter(lc);
+}
+```
+
+The `createLifecycle` could be very similar to the existing implementation. Alternatively, the original `single-spa-angular` is kept in place, and instead the `makeConverter` is called implicitly in these scenarios.
+
+Finally, the use of some APIs such as `navigate` needs to be adjusted.
+
+## Common practices (not enforced)
+
+With the new change nested / (React) routers in the frontend modules don't need to be created. The routing context remains the same, as the (React) components are running in the same React tree.
+
+Another new possibility is that the styleguide can now exclusively commit to React component. Things such as the notifications won't require a "wrapper" or "host" anymore, but can directly be attached to the main React tree by the app shell.
```

Comment:
Its just a possibility. And the given example (notifications) is a simple implementation change; right now its already using React, but does a `render`. Now it could just stay a component and be mounted by the app shell, which makes the whole model more consistent.